### ISSUE TITLE
[HOLD for payment 2023-03-17] [HOLD #14549] The country field is set to null while selecting US addresses, only in Spanish language preference

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#15014]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Go to Settings and change to the Spanish language
2. Go to Profile
3. Go to Personal information
4. Go to address
5. Try to first change the country to a different country besides the United states .
6. Next, change the location address to any US address. (like texas /arizona )  and Notice that the Country field is set to null and throws error while saving.

## Expected Result:
Country should be updated while selecting US address even in spanish

## Actual Result:
Country field is set to null and throws error while saving in spanish.

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.68-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/217963809-7c57aa3b-5116-4051-ba0f-fefd6809c08f.mov

https://user-images.githubusercontent.com/43996225/217964102-1307a0a8-94ba-4b5f-96a8-884a2f289705.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @ashimsharma10
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1675938847425899

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @abekkala (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by abekkala** [comment]
@Beamanator Would the fix for https://github.com/Expensify/App/issues/14549 resolve this one as well? 
Or should I continue on with this one as it's own Upwork job?

**Comment by abekkala** [comment]
I'm going to verify this needs it's own job first

**Comment by Beamanator** [comment]
Ooh good point @abekkala - that's verryyyy possible!

**Comment by Beamanator** [comment]
I do think it makes sense to hold this one for a bit - and I'll mention that we want to make this "fix" / "feature" via the fix for that other issue 👍

**Comment by abekkala** [comment]
Sounds good @Beamanator

**Comment by Beamanator** [comment]
Making weekly while this is on hold

**Comment by abekkala** [comment]
Still on hold for the other

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.81-1 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/15507


If no regressions arise, payment will be issued on 2023-03-17. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@Beamanator] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@Beamanator] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@Beamanator] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@abekkala] Determine if we should create a regression test for the bug.
- [ ] [@Beamanator] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to the [appropriate location](https://github.com/Expensify/App/blob/main/contributingGuides/TestRail_Navigation_C%2B.md) to ensure the same bug will not reach production again.
- [ ] [@abekkala] Review the proposed regression test steps and location.
- [ ] [@abekkala] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by MelvinBot** [comment]
Looks like something related to `react-navigation` may have been mentioned in this issue discussion.

As a reminder, please make sure that all proposals are not workarounds and that any and all attempt to fix the issue holistically have been made before proceeding with a solution. Proposals to change our `DeprecatedCustomActions.js` files should not be accepted.

Feel free to drop a note in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q) with any questions.

**Comment by abekkala** [comment]
@Beamanator - I'm confused by the latest Melvin posts lol
Was this issue fixed by the other GH or should we move forward fro this one to have it's own job?

**Comment by Beamanator** [comment]
Aah rightttt I thinkkk we can just check that this issue no longer exists in production, then we can probably close it out since payment & other BZ tasks are happening in https://github.com/Expensify/App/issues/14549

**Comment by ashimsharma10** [comment]
@Beamanator 
I'm the external reporter and have not paid for this. I'm a new member and don't know where to ask for it.

**Comment by MelvinBot** [comment]
📣 @ashimsharma10! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by Beamanator** [comment]
I agree with what you're saying @abekkala 👍
